lychee-lib/src/client.rs

```diff
@@ -650,13 +668,52 @@ impl Client {
     }
 
     /// Check a `file` URI.
-    pub fn check_file(&self, uri: &Uri) -> Status {
-        if let Ok(path) = uri.url.to_file_path() {
-            if path.exists() {
-                return Status::Ok(StatusCode::OK);
+    pub async fn check_file(&self, uri: &Uri) -> Status {
+        let Ok(path) = uri.url.to_file_path() else {
+            return ErrorKind::InvalidFilePath(uri.clone()).into();
+        };
+        if !path.exists() {
+            return ErrorKind::InvalidFilePath(uri.clone()).into();
+        }
+        if self.include_fragments {
+            self.check_fragment(&path, uri).await
+        } else {
+            Status::Ok(StatusCode::OK)
+        }
+    }
+
+    /// Checks a `file` URI's fragment.
```

Comment:
Encapsulating this logic sounds is a great idea. I renamed the `FragmentCache` to `FragmentChecker` because it isn't just a cache; it also parses files and populates the cache if it is entry is vacant.

I've left `extract_markdown_fragments` in `extract::markdown` because I like the idea of all the markdown parsing logic in one module. This would also be nice if we ever want a function that can parse for both links and fragments at the same time in the future.